runtime/os_interface/linux/drm_neo.cpp

```diff
@@ -132,10 +132,11 @@ bool Drm::is48BitAddressRangeSupported() {
 }
 
 void Drm::checkPreemptionSupport() {
-#if defined(I915_PARAM_HAS_PREEMPTION)
+#if defined(I915_SCHEDULER_CAP_PREEMPTION)
     int value = 0;
-    auto ret = getParam(I915_PARAM_HAS_PREEMPTION, &value);
-    preemptionSupported = (ret == 0 && value == 1);
+    auto ret = getParamIoctl(I915_PARAM_HAS_SCHEDULER, &value);
+    UNRECOVERABLE_IF(ret != 0);
```

Comment:
what about kernels that doesn't have I915_PARAM_HAS_SCHEDULER ?

Reply:
i915_drm.h:
#define I915_PARAM_HAS_SCHEDULER 41
#define I915_SCHEDULER_CAP_ENABLED (1ul << 0)
#define I915_SCHEDULER_CAP_PRIORITY (1ul << 1)
#define I915_SCHEDULER_CAP_PREEMPTION (1ul << 2)

This block of code is controlled by 'I915_SCHEDULER_CAP_PREEMPTION', according to the rules of kernel, if 'I915_SCHEDULER_CAP_PREEMPTION' is defined, the 'I915_PARAM_HAS_SCHEDULER' must be defined too.
If 'I915_PARAM_HAS_SCHEDULER' is not defined, then the total block won't execute and 'preemptionSupported' defaults to 'false'.

Reply:
In Neo we are using a copy of i915_drm.h located in 
vpg-compute-neo\third_party\uapi\drm\i915_drm.h

So this cap will always be present, but when code is executed on kernels that doesn't have this cap IOCTL will fail.

We shouldn't check for defines presence.

Reply:
thank you, understood.